pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1991,6 +2024,22 @@
       ^-  ?
       &(=(item-binding binding) =(item-duct duct))
     ==
+  ::  +add-cache: remember a URL->scry mapping
+  ::
+  ++  add-cache
+    |=  [url=@t =beam]
+    ^-  [(list move) server-state]
+    =.  cache.state  (~(put by cache.state) url beam)
+    =/  =riff:clay  [q.beam `[%next %x r.beam s.beam]]
+    :_  state
+    [duct %pass /watch-cache/[url] %c %warp our riff]~
+  ::  +del-cache: forget a URL->scry mapping
+  ::
+  ++  del-cache
+    |=  url=@t
+    ^-  [(list move) server-state]
+    =.  cache.state  (~(del by cache.state) url)
+    `state
```

Comment:
note that this doesn't immediately cancel the existing clay subscription. `watch-cache` will discard events that don't have a matching cache entry, so the subscription won't be renewed. But if we're worried about leaking subscriptions, we could explicitly cancel here.